src/components/Shop/Shop.jsx

```diff
@@ -1,37 +1,108 @@
-import React, { useState, useEffect } from 'react';
-import { useQuery } from '@apollo/react-hooks';
+import React from 'react';
+import { useQuery, useMutation } from '@apollo/react-hooks';
+import store from 'store';
+import sumBy from 'lodash/sumBy';
 import { Link } from 'react-router-dom';
 import Card from '@material-ui/core/Card';
 import CardActionArea from '@material-ui/core/CardActionArea';
 import CardActions from '@material-ui/core/CardActions';
 import CardContent from '@material-ui/core/CardContent';
 import CardMedia from '@material-ui/core/CardMedia';
-import store from 'store';
 import Button from '@material-ui/core/Button';
 import Typography from '@material-ui/core/Typography';
 import Grid from '@material-ui/core/Grid';
 import useStyles from './Styles';
 import Spinner from '../Spinner';
-import { GET_PRODUCTS } from './GraphQl';
+import {
+  GET_PRODUCTS,
+  CREATE_CHECKOUT,
+  GET_CHECKOUT_ID,
+  CHECKOUT_LINE_ITEMS_REPLACE
+} from './GraphQl';
+import { GET_CHECKOUT_ITEMS } from '../Cart/GraphQL';
 import noPhotoAvailable from '../../media/noPhotoAvailable.png';
 
 export default function Shop() {
   const classes = useStyles();
-  const [productsIds, setProductsIds] = useState(store.get('ids') || []);
-  const { loading, data: { products } = {}, client } = useQuery(GET_PRODUCTS);
+  const {
+    data: { checkoutId }
+  } = useQuery(GET_CHECKOUT_ID);
+  const { loading, data: { products } = {} } = useQuery(GET_PRODUCTS);
 
-  const addToCart = id => {
-    const listProducts = store.get('ids') || [];
-    setProductsIds([...listProducts, id]);
-  };
+  const [checkoutCreate, { loading: checkoutCreateLoad, client }] = useMutation(
+    CREATE_CHECKOUT,
+    {
+      onCompleted: data => {
+        const cartItems = data.checkoutCreate.checkout.lineItems.edges.map(
+          ({ node }) => ({
+            variantId: node.variant.id,
+            quantity: node.quantity
+          })
+        );
+        store.set('checkoutId', data.checkoutCreate.checkout.id);
+        store.set('cartItems', cartItems);
+        client.writeData({
+          data: {
+            checkoutId: data.checkoutCreate.checkout.id
+          }
+        });
+      }
+    }
+  );
+  const [
+    checkoutLineItemsReplace,
+    { loading: checkoutReplaceLoad }
+  ] = useMutation(CHECKOUT_LINE_ITEMS_REPLACE, {
+    refetchQueries: [
+      {
+        query: GET_CHECKOUT_ITEMS,
+        variables: { id: store.get('checkoutId') }
+      }
+    ],
+    onCompleted: data => {
+      const cartItems = data.checkoutLineItemsReplace.checkout.lineItems.edges.map(
```

Comment:
This functionality duplicated in two files. I think you can easily move this flow to some action or service.
The duplicate is contained in this file -> src/components/LineItem/LineItem.jsx